qiskit/qasm3/ast.py

```diff
@@ -114,6 +114,27 @@ def __init__(self):
         pass
 
 
+class ClassicalType(ASTNode):
+    """Information about a classical type.  This is just an abstract base for inheritance tests."""
+
+
+class FloatType(ClassicalType, enum.IntEnum):
+    """Allowed values for the width of floating-point types."""
+
+    half = 16
+    single = 32
+    double = 64
+    quad = 128
+    oct = 256
+
+
+class BitArrayType(ClassicalType):
+    """Type information for a sized number of classical bits."""
+
+    def __init__(self, size: int):
+        self.size = size
+
+
```

Comment:
I only added the types that were actually required for things to work correctly to try and keep the line count down a little.